api/types/app.go

```diff
@@ -270,6 +271,19 @@ func (a *AppV3) CheckAndSetDefaults() error {
 	if a.Spec.URI == "" {
 		return trace.BadParameter("app %q URI is empty", a.GetName())
 	}
+
+	url, err := url.Parse(a.Spec.PublicAddr)
+	if err != nil {
+		return trace.BadParameter("invalid PublicAddr format: %v", err)
+	}
+	host := a.Spec.PublicAddr
+	if url.Host != "" {
+		host = url.Host
+	}
+	if strings.HasPrefix(host, constants.KubeSNIPrefix) {
+		return trace.BadParameter("app %q DNS prefix found in %q public_url is reserved for internal usage",
+			constants.KubeSNIPrefix, a.Spec.PublicAddr)
+	}
```

Comment:
just trying to understand what we want to achieve here. so basically `kube.` is reserved that it CANNOT be part of the public addr?  Do we have documentation somewhere so people can expect this? I think it's reasonable if someone wants to have a cluster with public addr `kube.xxxx.xxx`

Reply:
Yes, this behavior is undocumented but "kube." prefix is used internally to route requests to Kube service instead of Teleport web service. Right now if a user will set public address with kube prefix the TLS handshake will fail because because it will be terminated agains HostCA instead of Web CA. 
 
 Probably it is reasonable to move from kube prefix to more uncommon `prefix` like `kube-server-name-prefix`

Reply:
I was just curious if we have any docs on this. Like AWS usually list their reserved words. I think it's ok as long as we don't loose track. Like this error message is already a lot better than hitting this bug. =D